kin-sdk-core/src/main/java/kin/sdk/core/EncryptorFactory.java

```diff
@@ -0,0 +1,45 @@
+package kin.sdk.core;
+
+import android.content.Context;
+import android.os.Build;
+import android.os.Build.VERSION_CODES;
+
+class EncryptorFactory {
+
+    private static final String VERSION_KEY = "encryptor_ver";
+    private static final String VER_23 = "23";
+    private static final String VER_18 = "18";
+    private static final String VER_16 = "16";
+
+    //no instances
+    private EncryptorFactory() {
+    }
+
+    static Encryptor create(Context context, Store store) {
+        String versionString = store.getString(VERSION_KEY);
```

Comment:
you can simplify the logic here by having a separate method that returns the version to be used.

Then you can simply switch over the version to decide which encryptor to use

Reply:
no option to upgrade encryptor - let product know